crates/ty_server/src/server/api/requests/workspace_diagnostic.rs

```diff
@@ -27,9 +27,7 @@ impl BackgroundRequestHandler for WorkspaceDiagnosticRequestHandler {
         _client: &Client,
         _params: WorkspaceDiagnosticParams,
     ) -> Result<WorkspaceDiagnosticReportResult> {
-        let index = snapshot.index();
-
-        if !index.global_settings().diagnostic_mode().is_workspace() {
+        if !snapshot.global_settings().diagnostic_mode().is_workspace() {
```

Comment:
What would you think of naming `global_settings`, `window_settings` (I'm okay with global settings. I'm just wondering if it's worth aligning with VS code's terminology)